src/improvements/AddressRegistry.sol

```diff
@@ -413,31 +413,27 @@ contract AddressRegistry is IAddressRegistry, Test {
         view
         returns (address)
     {
-        try IFetcher(faultDisputeGame).anchorStateRegistry() returns (address anchorStateRegistryProxy) {
-            return anchorStateRegistryProxy;
-        } catch {
-            return IFetcher(permissionedDisputeGame).anchorStateRegistry();
-        }
+        return IFetcher(permissionedDisputeGame).anchorStateRegistry();
     }
 
     function getDelayedWETHProxy(address faultDisputeGame, address permissionedDisputeGame)
         internal
         view
         returns (address)
     {
-        try IFetcher(faultDisputeGame).weth() returns (address delayedWethProxy) {
-            return delayedWethProxy;
-        } catch {
-            return IFetcher(permissionedDisputeGame).weth();
-        }
+        (bool ok, bytes memory data) =
+            address(faultDisputeGame).staticcall(abi.encodeWithSelector(IFetcher.weth.selector));
+        if (ok && data.length == 32) return abi.decode(data, (address));
+
+        return IFetcher(permissionedDisputeGame).weth();
     }
 
     function getMips(address faultDisputeGame, address permissionedDisputeGame) internal view returns (address) {
-        try IFetcher(faultDisputeGame).vm() returns (address mips) {
-            return mips;
-        } catch {
-            return IFetcher(permissionedDisputeGame).vm();
-        }
+        (bool ok, bytes memory data) =
+            address(faultDisputeGame).staticcall(abi.encodeWithSelector(IFetcher.vm.selector));
+        if (ok && data.length == 32) return abi.decode(data, (address));
+
+        return IFetcher(permissionedDisputeGame).vm();
     }
```

Comment:
Similar to the ASR method, no reason to check the FDG and take that as an input here

Reply:
https://github.com/ethereum-optimism/superchain-ops/pull/582/commits/d706fec9ae0caf08656b793d2b1d3d96b297ed8b